go-update/src/test/java/org/reactome/release/goupdate/JavaVsPerlComparison.java

```diff
@@ -82,18 +127,22 @@ public int compare(GKInstance o1, GKInstance o2)
 				MySQLAdaptor perlDBAdaptor ;
 				if (!perlAdaptorPool.containsKey(threadID))
 				{
-					perlDBAdaptor = new MySQLAdaptor(perlUpdatedDB.getDBHost(), perlUpdatedDB.getDBName(), perlUpdatedDB.getDBUser(), perlUpdatedDB.getDBPwd(), perlUpdatedDB.getDBPort());
+					perlDBAdaptor = new MySQLAdaptor(
+						perlUpdatedDB.getDBHost(), perlUpdatedDB.getDBName(), perlUpdatedDB.getDBUser(),
+						perlUpdatedDB.getDBPwd(), perlUpdatedDB.getDBPort()
+					);
 					perlAdaptorPool.put(threadID, perlDBAdaptor);
 				}
 				else
 				{
 					perlDBAdaptor = perlAdaptorPool.get(threadID);
 				}
-				
+
 				javaGoInst.setDbAdaptor(javaDBAdaptor);
 				String accession = (String) javaGoInst.getAttributeValue(ReactomeJavaConstants.accession);
 				@SuppressWarnings("unchecked")
-				Collection<GKInstance> perlGoInsts = (Collection<GKInstance>) perlDBAdaptor.fetchInstanceByAttribute(javaGoInst.getSchemClass().getName(), ReactomeJavaConstants.accession, "=", accession);
+				Collection<GKInstance> perlGoInsts = accessionToPerlInstances.get(accession);
```

Comment:
Not sure I'm reading this correctly, but it seems like an awful lot of new code was added to populate `accessionToPerlInstances`. Does it have a different result than the old code?

Reply:
It allows the GO instances from the Perl database to be queried at once and stored in a map of String accession value to the corresponding GKInstance.  The old code was making one database call for each individual accession.  When I was refactoring, I noticed a slower running time and I made this change to try to address it.  I think the issue may have been somewhere else in the way I structured something, but I thought it would be good to keep this change to have a single bulk query to the database rather than individual ones.